src/lib/Handshake.js

```diff
@@ -194,79 +202,80 @@ class Handshake {
     return nonce;
   };
 
-  // TODO wrong method name? it read deviceID alongside with
-  // deviceKey? and returns deviceProvidedPem
-  _readDeviceID = (nonce: Buffer, data: Buffer): ?string => {
-    //server should read 256 bytes
-    //decrypt msg using server private key
-    let plaintext;
-    try {
-      plaintext = CryptoLib.decrypt(CryptoLib.getServerKeys(), data);
-    } catch (error) {
-      logger.error(`Handshake decryption error: ${error}`);
-    }
+  _readDeviceHandshakeData = (nonce: Buffer, data: Buffer): {
+    deviceID: string,
+    deviceProvidedPem: ?string,
+  } => {
+    const decryptedHandshakeData = CryptoLib.decrypt(CryptoLib.getServerKeys(), data);
```

Comment:
Bleh.. I think `ICrypto` needs to be cleaned up. I missed it with all my flow types... I'm sure there is plenty of room for refactoring too. It should also be renamed to `CryptoManager`